include/openPMD/backend/Attributable.hpp

```diff
@@ -190,12 +190,10 @@ class Attributable
 
     std::shared_ptr< Writable > m_writable;
     /* views into the resources held by m_writable
-     * purely for convenience so code that uses these does not have to go through m_wriable-> */
+     * purely for convenience so code that uses these does not have to go through m_writable-> */
     AbstractFilePosition* abstractFilePosition;
     AbstractIOHandler* IOHandler;
     Writable* parent;
-    bool& dirty;
-    bool& written;
```

Comment:
Those are the main problem.
Alternative approaches:
- expose as public member function (maybe)
- expose as public pointer member (urgh)